igvc_navigation/src/mapper/main.cpp

```diff
@@ -95,78 +136,46 @@ void frame_callback(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr &msg, const s
+
+  // TODO: time should be based on message timestamp.
+  ros::Time time = ros::Time::now();
+  // Check if static transform already exists for this topic.
+  checkExistsStaticTransform(msg, topic);
+
+  // Apply transformation to msg points using the transform for this topic.
+  pcl_ros::transformPointCloud(*msg, *transformed, transforms.at(topic));
+  handlePointCloud(transformed);
 
-    // TODO: Clear up these variable names. x_point, x_robot, etc.
-    int point_x = static_cast<int>(std::round(x_loc / resolution + cur_x / resolution + start_x));
-    int point_y = static_cast<int>(std::round(y_loc / resolution + cur_y / resolution + start_y));
-    if (point_x >= 0 && point_y >= 0 && point_x < length_y && start_y < width_x)
-    {
-      // TODO: NO MAGIC NUMBERS.
-      // TODO: Have variables for probability thresholds/steps.
-      // TODO: Investigate decay, potentially use CV_16U, decrement at a certain rate.
-      // TODO: Potentially change
-      if (published_map->at<uchar>(point_x, point_y) < 230)
-      {
-        published_map->at<uchar>(point_x, point_y) += (uchar)125;
-      }
-      count++;
-    }
-    else if (!offMap)
-    {
-      ROS_WARN_STREAM("Some points out of range, won't be put on map.");
-      offMap = true;
-    }
-  }
   img_bridge = cv_bridge::CvImage(msgBoi.header, sensor_msgs::image_encodings::MONO8, *published_map);
   img_bridge.toImageMsg(imageBoi);  // from cv_bridge to sensor_msgs::Image
   time = ros::Time::now();          // so times are exact same
-  imageBoi.header.stamp = time;
-  msgBoi.header.stamp = time;
-  msgBoi.header.frame_id = "/odom";
-  msgBoi.image = imageBoi;
-  msgBoi.length = length_y;
-  msgBoi.width = width_x;
-  msgBoi.resolution = resolution;
-  msgBoi.orientation = orientation;
-  ROS_INFO_STREAM("robot location " << std::round(cur_x / resolution) + start_x << ", "
-                                    << std::round(cur_y / resolution) + start_y);
-  msgBoi.x = std::round(cur_x / resolution) + start_x;
-  msgBoi.y = std::round(cur_y / resolution) + start_y;
-  msgBoi.x_initial = start_x;
-  msgBoi.y_initial = start_y;
+  setMsgValues(time);
+  ROS_INFO_STREAM("robot location " << std::round(x_robot / resolution) + start_x << ", "
+                                    << std::round(y_robot / resolution) + start_y);
   map_pub.publish(msgBoi);
   if (debug)
   {
     debug_pub.publish(imageBoi);
-    // ROS_INFO_STREAM("\nThe robot is located at " << cur_x << "," << cur_y << "," << orientation);
+    // ROS_INFO_STREAM("\nThe robot is located at " << x_robot << "," << y_robot << "," << orientation);
     pcl::PointCloud<pcl::PointXYZRGB>::Ptr fromOcuGrid =
         pcl::PointCloud<pcl::PointXYZRGB>::Ptr(new pcl::PointCloud<pcl::PointXYZRGB>());
     for (int i = 0; i < width_x; i++)
     {
       for (int j = 0; j < length_y; j++)
       {
-        if (published_map->at<uchar>(i, j) >= (uchar)178)
+        if (published_map->at<uchar>(i, j) >= (uchar)occupancy_grid_threshold)
```

Comment:
I would cast occupancy_grid_threshold to uchar and then direct compare instead of doing this each time